Automattic/jetpack
# Add a conditional recommendation about security

#### Changes proposed in this Pull Request:

* This PR adds a new recommendation step in the Recommendations interface that is conditionally enabled when a new plugin is activated.
* The recommendation provides some more information about plugin security and links to an article about WordPress security.

#### Jetpack product discussion


pbNhbs-1iw-p2

#### Does this pull request change what data or activity we track or use?


Yes, this PR can record following new events when interacting with the resource recommendation step, including the slug of the step as a property of the event.
```
jetpack_recommended_resource_learn_more_click
jetpack_recommended_resource_read_click
jetpack_recommended_resource_skip_click
```
The new resource recommendation is also included on the summary screen and records the following new event when the "Learn More" button is clicked, passing the feature name as a property of the event.
```
jetpack_recommendations_summary_learn_more_click
```

#### Testing instructions:






To reset the assistant state while testing, you can use the following commands ( you can run these with `jetpack docker` if using the Docker development environment )
* `wp jetpack options delete recommendations_data`
* `wp jetpack options delete recommendations_conditional`

If your test site already has a plan or product(s) that provides Backup, Scan and Anti-Spam, you will not receive the new recommendation as outlined in the test plan below. Try testing using a site that has a Jetpack free plan.

---

* Checkout this branch on your Jetpack test site.
* Visit the Jetpack dashboard. Note that the "Recommendations" tab in the dashboard navigation does not show any badge indicating that there is a new recommendation.
![Screen Shot 2022-03-21 at 2 27 58 PM](https://user-images.githubusercontent.com/18016357/159340092-2adb1a7e-7834-4986-b965-33519fa1ee11.png)

* Now, go to the plugins page and enable the "Hello Dolly" plugin. You can disable it again right away if you like.
* After the enabling the plugin, navigate back to the Jetpack Dashboard. You should see a red badge with a "1" next to the Recommendations tab in the navigation on the Jetpack dashboard - indicating a new recommendation is available.
![Screen Shot 2022-03-11 at 4 00 23 PM](https://user-images.githubusercontent.com/18016357/158703217-3b7bb9bb-a1e9-4b6a-814e-d81e167094f1.png)

* Click on the "Recommendations" tab in the navigation - you should now see a recommendation step that provides some information about plugin security. There should be a green "New" badge that shows on this step in the upper-left corner. The CTA button should link out to this article: https://jetpack.com/2021/10/15/wordpress-security-for-beginners-2/

![Screen Shot 2022-03-24 at 11 46 38 AM](https://user-images.githubusercontent.com/18016357/160194644-75799bc7-87d5-4cfb-9c27-21909201954e.png)

* Click on "Read Later"  to proceed to the summary screen. On the summary screen, you should see the a new "Resources" section with the "Site Security" recommendation listed. A green "New" badge should show next to "Site Security" in the list. The "Read More" button should link to the same article mentioned in the previous step.

![Screen Shot 2022-03-24 at 11 46 53 AM](https://user-images.githubusercontent.com/18016357/160194716-552e1104-8b62-48b7-a9e4-895051413257.png)

* If you reload the page on the summary screen, the green "New" badge next to Site Security in the summary resources list and the red "1" indicator next to "Recommendations" in the dashboard navigation  should no longer show.
* Automated unit tests should be passing
* Visuals should match this reference: G151IYl8Lm4f0NjgLpR943-fi-2132%3A39570
